src/main/java/net/minecraft/launchwrapper/protocol/SkinURLConnection.java

```diff
@@ -0,0 +1,157 @@
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import javax.imageio.ImageIO;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.awt.image.BufferedImage;
+import java.awt.*;
+import java.util.Map;
+import java.net.Proxy;
+
+import com.mojang.authlib.minecraft.MinecraftProfileTexture;
+import com.mojang.authlib.ProfileLookupCallback;
+import com.mojang.authlib.GameProfile;
+import com.mojang.authlib.Agent;
+import com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService;
+
+public class SkinURLConnection extends HttpURLConnection {
+    public SkinURLConnection(URL url) {
+        super(url);
+    }
+
+    @Override
+    public void disconnect() {
+    }
+
+    @Override
+    public boolean usingProxy() {
+        return false;
+    }
+
+    InputStream inputStream = null;
+    int responseCode = 200;
+
+    @Override
+    public void connect() throws IOException {
+        String url = this.url.toString();
+        String username = (url.contains("/MinecraftSkins/")
+                ? url.substring(url.indexOf("/MinecraftSkins/"))
+                .replace("/MinecraftSkins/", "")
+                .replace(".png", "")
+                : url.substring(url.indexOf("/skin/")))
+                .replace("/skin/", "")
+                .replace(".png", "");
+
+        try {
+            MinecraftProfileTexture skin = getUserSkin(username);
+            boolean slim = "slim".equals(skin.getMetadata("model"));
+            inputStream = convertModernSkin(new URL(skin.getUrl()), slim);
+        } catch (Exception ex) {
+            responseCode = 404;
+        }
+    }
+
+    GameProfile gameProfile = null;
+    YggdrasilAuthenticationService authenticationService = new YggdrasilAuthenticationService(Proxy.NO_PROXY, (String)null);
+
+    private MinecraftProfileTexture getUserSkin(String username) {
+        authenticationService.createProfileRepository().findProfilesByNames(new String[] { username }, Agent.MINECRAFT, new ProfileLookupCallback() {
+            public void onProfileLookupSucceeded(GameProfile paramGameProfile) {
+                gameProfile = paramGameProfile;
+            }
+            public void onProfileLookupFailed(GameProfile paramGameProfile, Exception paramException) {
+            }
+        });
+
+        if (gameProfile == null)
+            return null;
+
+        gameProfile = authenticationService.createMinecraftSessionService().fillProfileProperties(gameProfile, true);
+
+        Map<MinecraftProfileTexture.Type, MinecraftProfileTexture> textures = authenticationService.createMinecraftSessionService().getTextures(gameProfile, true);
+
+        if (textures.containsKey(MinecraftProfileTexture.Type.SKIN))
+            return textures.get(MinecraftProfileTexture.Type.SKIN);
+
+        return null;
+    }
+
+    public static InputStream convertModernSkin(URL skinUrl, boolean slim) throws IOException {
```

Comment:
developer note:
This function stretches and flattens the modern day skins to suite the old-style Steve skins. It is optional and performant (only called once per user in a game).
In the screenshot I've attached to this PR, I am using an Alex skin, you can see it stretched out to suite Steve's larger arms. Obviously this isn't perfect, but it's realistically the best you can get with old versions.